packages/aws-cdk/test/api/hotswap/appsync-mapping-templates-hotswap-deployments.test.ts

```diff
@@ -0,0 +1,308 @@
+  });
+});
+
+test('does not call the updateResolver() API when it receives a change that is not a mapping template difference in a Resolver', async () => {
+  // GIVEN
+  setup.setCurrentCfnStackTemplate({
+    Resources: {
+      AppSyncResolver: {
+        Type: 'AWS::AppSync::Resolver',
+        Properties: {
+          RequestMappingTemplate: '## original template',
+          FieldName: 'oldField',
+        },
+        Metadata: {
+          'aws:asset:path': 'old-path',
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        AppSyncResolver: {
+          Type: 'AWS::AppSync::Resolver',
+          Properties: {
+            RequestMappingTemplate: '## new template',
+            FieldName: 'newField',
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await hotswapMockSdkProvider.tryHotswapDeployment(cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).toBeUndefined();
+  expect(mockUpdateResolver).not.toHaveBeenCalled();
+});
+
+test('does not call the updateResolver() API when a resource with type that is not AWS::AppSync::Resolver but has the same properties is changed', async () => {
+  // GIVEN
+  setup.setCurrentCfnStackTemplate({
+    Resources: {
+      AppSyncResolver: {
+        Type: 'AWS::AppSync::NotAResolver',
+        Properties: {
+          RequestMappingTemplate: '## original template',
+          FieldName: 'oldField',
+        },
+        Metadata: {
+          'aws:asset:path': 'old-path',
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        AppSyncResolver: {
+          Type: 'AWS::AppSync::NotAResolver',
+          Properties: {
+            RequestMappingTemplate: '## new template',
+            FieldName: 'newField',
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await hotswapMockSdkProvider.tryHotswapDeployment(cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).toBeUndefined();
+  expect(mockUpdateResolver).not.toHaveBeenCalled();
+});
+
+test('calls the updateFunction() API when it receives only a mapping template difference in a Function', async () => {
+  // GIVEN
+  setup.setCurrentCfnStackTemplate({
+    Resources: {
+      AppSyncFunction: {
+        Type: 'AWS::AppSync::FunctionConfiguration',
+        Properties: {
+          Name: 'my-function',
+          ApiId: 'apiId',
+          DataSourceName: 'my-datasource',
+          FunctionVersion: '2018-05-29',
+          RequestMappingTemplate: '## original request template',
+          ResponseMappingTemplate: '## original response template',
+        },
+        Metadata: {
+          'aws:asset:path': 'old-path',
+        },
+      },
+    },
+  });
+  setup.pushStackResourceSummaries(setup.stackSummaryOf('AppSyncFunction', 'AWS::AppSync::FunctionConfiguration', 'my-function'));
```

Comment:
So here, this should _not_ be needed (notice that the template contains the `Name` property).